### PR TITLE
Add static commands to documentation

### LINKED ISSUE #10970: Add all commands (including those without keymaps) to documentation
There are a number of commands ([here](https://github.com/helix-editor/helix/blob/dbacaaddcaa4a9dd39c73d0102d03ea7c24ca647/helix-term/src/commands.rs#L250-L534)) such as `select_line_above` that can be used in custom keymaps but are (as far as I can tell) not formally documented, so the only way to determine which commands can be used is to read the source code or possibly from perusing the changelog. It would be nice to have these commands documented somewhere, e.g. at the end of https://docs.helix-editor.com/keymap.html under a new section that contains all of those commands that are not bound by default to a keymap but that can be used in `config.toml`.

### PR DESCRIPTION
**Pull request by david-crespo**
Picking up https://github.com/helix-editor/helix/pull/11699 (itself a redo of https://github.com/helix-editor/helix/pull/11152) with [permission](https://github.com/helix-editor/helix/pull/11699#issuecomment-2439976390) from @adamperkowski. Closes #10970.

### CODE REVIEW

**Comment by david-crespo** [comment]
Heh, the docgen test fails because there is some nondeterministic ordering in the generated stuff. Will fix.

File: xtask/src/docgen.rs [hunk 1]
```diff
@@ -48,6 +54,65 @@ pub fn typable_commands() -> Result<String, DynError> {
...
+                            .map(|bind| {
+                                let keys = &bind
+                                    .iter()
+                                    .map(|key| key.key_sequence_format())
+                                    .collect::<String>();
+                                format!("`` {} ``", keys)
+                            })
+                            .collect();
+                        // sort for stable output. sort simple letters-only binds first, then lexicographic
+                        bind_strings.sort_by_key(|s| (s.contains('<'), s.to_owned()));
```
Comment (david-crespo) [inline xtask/src/docgen.rs:90]: Let me know if this is too cute. It makes sure the `h` is first here.

```md
| `move_char_left` | Move left | normal: `` h ``, `` <left> ``, insert: `` <left> `` |
| `move_char_right` | Move right | normal: `` l ``, `` <right> ``, insert: `` <right> `` |
```

**Comment by adamperkowski** [comment]
Yeah it's because of `HashMap`. Had a similar issue with a parser before, got rid of it entirely.

**Comment by david-crespo** [comment]
Bump for @the-mikedavis since you can just merge stuff now 🙂

File: xtask/src/docgen.rs (hunk 1 above)
Comment (the-mikedavis) [reply xtask/src/docgen.rs:90]: I think sorting by `s.len()` might make more sense? The byte length should be a more generic proxy for "shorter command sequence" than `.contains('<')` and might more effectively group similar binds like `page_up`

File: book/src/commands.md [hunk 2]
```diff
@@ -1,5 +1,16 @@
 # Commands
 
-Command mode can be activated by pressing `:`. The built-in commands are:
+- [Typable commands](#typable-commands)
+- [Static commands](#static-commands)
+
+## Typable commands
+
+Typable commands are used from command mode and may take arguments. Command mode can be activated by pressing `:`. The built-in commands are:
```
Comment (the-mikedavis) [inline book/src/commands.md:8]:
```suggestion
Typable commands are used from command mode and may take arguments. Command mode can be activated by pressing `:`. The built-in typable commands are:
```

File: book/src/generated/static-cmd.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+| `dap_variables` | List variables | normal: `` <space>Gv ``, select: `` <space>Gv `` |
+| `dap_terminate` | End debug session | normal: `` <space>Gt ``, select: `` <space>Gt `` |
+| `dap_edit_condition` | Edit breakpoint condition on current line | normal: `` <space>G<C-c> ``, select: `` <space>G<C-c> `` |
+| `dap_edit_log` | Edit breakpoint log message on current line | normal: `` <space>G<C-l> ``, select: `` <space>G<C-l> `` |
+| `dap_switch_thread` | Switch current thread | normal: `` <space>Gst ``, select: `` <space>Gst `` |
+| `dap_switch_stack_frame` | Switch stack frame | normal: `` <space>Gsf ``, select: `` <space>Gsf `` |
+| `dap_enable_exceptions` | Enable exception breakpoints | normal: `` <space>Ge ``, select: `` <space>Ge `` |
+| `dap_disable_exceptions` | Disable exception breakpoints | normal: `` <space>GE ``, select: `` <space>GE `` |
+| `shell_pipe` | Pipe selections through shell command | normal: `` | ``, select: `` | `` |
+| `shell_pipe_to` | Pipe selections into shell command ignoring output | normal: `` <A-|> ``, select: `` <A-|> `` |
```
[the remark this answers is not in the document]
Comment (david-crespo) [reply book/src/generated/static-cmd.md:282]: I tried the `<code>` thing, but as you say, that introduces new escaping issues: you end up playing whack-a-mole escaping every character that is significant in markdown, like `*` and <code>\`</code> and `~`. But then I tried the `\\|` approach and it seems to work fine. What was the problem you were seeing?

image

[the remark this answers is not in the document]
Comment (the-mikedavis) [reply book/src/generated/static-cmd.md]: Oh hmm I thought I saw GitHub's markdown renderer reject it but it seems to work:

| Column A | Column B |
|----------|----------|
| `shell_pipe` | `` \| `` |

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**